vignettes/complete-multiverse-analysis.Rmd

```diff
@@ -232,32 +233,137 @@ Since the multiverse object has already been created and the one parameter has a
   head()
 ```
+
+
+## Specifying conditions in the multiverse analysis
+In a multiverse analysis, it may occur that the value of one variable might depend on the value of another variable defined previously. 
+For example, in our example, we are excluding participants based on their *cycle length*. This can be done in two ways: we can use the values of the variable,`ComputedCycleLength` or `ReportedCycleLength`. If we are using `ComputedCycleLength` to exclude participants, this means that we should not calculate the variable `NextMenstrualOnset` (date for the onset of the next menstrual cycle) using the `ReportedCycleLength` value. Similarly, if we are using `ReportedCycleLength` to exclude participants it is inconsistent to calculate `NextMenstrualOnset` using `ComputedCycleLength`.
+
+We should be able to express these conditions in the multiverse. We can do this in two ways:
+1. `%when%`: when declaring a branch, we can use this operator to specify the condition as `A %when% B`
+2. `branch_assert`: the conditional $A | B$ is also referred to as the connective $A \implies B$. This has the meaning "if A is true, then B is also true" and is an abbreviation for $\neg A | B$
+
+#### The %when% operator
+There are two ways in which you can specify the `%when%` operator. The first is to specify it at the end of the branch. This will work even if you omit the branch option name.
+
+```{r, eval = FALSE}
+M$df <- data.raw.study2  %>%
+    mutate( ComputedCycleLength = StartDateofLastPeriod - StartDateofPeriodBeforeLast ) %>%
+    mutate(NextMenstrualOnset = branch(menstrual_calculation,
+            "mc_option1" ~ (StartDateofLastPeriod + ComputedCycleLength) %when% (cycle_length != "cl_option3"),
+            "mc_option2" ~ (StartDateofLastPeriod + ReportedCycleLength) %when% (cycle_length != "cl_option2"),
+            "mc_option3" ~ StartDateNext)
+    )
+```
+
+The other is to specify it at the head of the branch, right after the option name:
+
+```{r, eval = FALSE}
+M$df <- df  %>%
+    mutate(NextMenstrualOnset = branch(menstrual_calculation,
+            "mc_option1" %when% (cycle_length != "cl_option3") ~ StartDateofLastPeriod + ComputedCycleLength,
+            "mc_option2" %when% (cycle_length != "cl_option2") ~ (StartDateofLastPeriod + ReportedCycleLength),
+            "mc_option3" ~ StartDateNext)
+    )
+```
+
+#### The `branch_assert()` function
+Within the branch assert function, the user needs to specify the classical logic specification $\neg A | B$:
+
+```{r, eval = FALSE}
+M = multiverse()
+M$df <- df  %>%
+    mutate(NextMenstrualOnset = branch(menstrual_calculation,
+            "mc_option1" ~ StartDateofLastPeriod + ComputedCycleLength,
+            "mc_option2" ~ StartDateofLastPeriod + ReportedCycleLength,
+            "mc_option3" ~ StartDateNext)
+    ) %>%
+    branch_assert(menstrual_calculation != "mc_option1" | cycle_length != "cl_option3") %>%
```

Comment:
The use of OR seems odd here. Wouldn't this cause *all* of the universes that have either of these options set to be excluded --- which would be equivalent to not specifying them in the first place?

Reply:
This one is equivalent to (I guess this is a better way of specifying the assertion):
!(menstrual_calculation == "mc_option1" & cycle_length == "cl_option3")

But it allows menstrual_calculation == "mc_option1"  when cycle_length == "cl_option1" or "cl_option2"